tools/docker

```diff
@@ -20,23 +20,18 @@ ROOT=$(dirname "$WD")
 
 set -eu
 
-TARGET_OS=${TARGET_OS:-linux}
-TARGET_ARCH=${TARGET_ARCH:-amd64}
-GOOS_LOCAL=${TARGET_OS}
-GOARCH_LOCAL=${TARGET_ARCH}
-
 cd "${ROOT}"
 
 export REPO_ROOT="${ROOT}"
 
-if [[ -f out/${TARGET_OS}_${TARGET_ARCH}/docker-builder ]]; then
+if [[ -f ${LOCAL_OUT}/docker-builder ]]; then
```

Comment:
can we keep the defaults? I would prefer this to be runnable directly since IMO the UX is better than the make variables.

Or is it not reasonable since we have so much logic to set up the vars? Perhaps we can do `source ./common/scripts/setup_env.sh envfile` or similar if they are not set?

Reply:
Problem is that the TARGET variables point to the HOST environment and in the case of using the build container on a Mac (any non-linux host), those values are Darwin/other values, and we need to build/run the docker-builder in the linux container. 

I could try looking to see if the LOCAL_OUT is set and use that (would have to parse to get OS/ARCH) and if not set fall back to the linux values. I'm not sure about moving LOCAL_OUT out to the setup_env.sh (where we set the TARGET*) and calling that, but that might actually be the best. Thoughts?

Reply:
Ah its not already there, is it...

It could be tricky since it depends on IN_BUILD_CONTAINER. maybe it would work though... 

Either way works fine for me.

Reply:
@howardjohn I tried to simply set TARGET_ARCH and TARGET_OS from the LOCAL_OUT. This resulted in the follow-on `make` having an issue with it thinking it needed to use the `linux` side of the `out` directory instead of the `darwin` side as the TARGET* env vars were overridden. I suspect that trying to run setup_env again will have the same effect. This isn't necessarily an elegant solution, but does set the variables to build docker-build and not affect the makefile call.